src/check.rs

```diff
@@ -298,6 +353,96 @@ fn offensive_cargo_config_check(path: &Path) -> bool {
     false
 }
 
+/// A combination of `get_version` and `prompt_install`: if the tool is not found
+/// or does not meet the minimum version (when provided) and an install command
+/// is provided, it will prompt the user to install/upgrade it and then re-check.
+fn get_version_or_install(
+    cmd: &str,
+    args: &[&str],
+    install_cmd: Option<&[&str]>,
+    min_version: Option<(u8, u8, u8)>,
+) -> Option<Version> {
+    let version = get_version(cmd, args);
+
+    if let Some((min_major, min_minor, min_patch)) = min_version {
+        match check_version(version.clone(), min_major, min_minor, min_patch) {
+            CheckResult::Ok(_) => {
+                // Installed and new enough – nothing to do
+                return version;
+            }
+            CheckResult::WrongVersion | CheckResult::NotFound => {
+                let Some(install_cmd) = install_cmd else {
+                    // no way to offer an automatic install/upgrade
+                    return version;
+                };
+                prompt_install(cmd, install_cmd);
+            }
+        }
+    } else if version.is_some() {
+        // We don't know minimum version and the tool exists – nothing to do
+        return version;
+    }
+
+    get_version(cmd, args)
+}
+
+fn prompt_install(name: &str, cmd: &[&str]) {
+    let command_str = cmd.join(" ");
+    println!("🛑 {name} is not installed or is below the required version.");
+
+    if name == "probe-rs" && cfg!(target_os = "linux") {
+        println!(
+            "💡 On Linux, probe-rs requires additional setup before installation.\n\
+            See https://probe.rs/docs/getting-started/installation/ for details."
+        );
+    }
+
+    println!("Do you want to run `{command_str}` now? [y/N]");
+
+    enable_raw_mode().unwrap();
```

Comment:
I don't think we should be unwrapping here, let's handle the error locally if we can or propagate the error so we can handle it gracefully and report a proper error